code/game/objects/items/weapons/butterfly.dm

```diff
@@ -0,0 +1,100 @@
+/obj/item/weapon/butterflyknife
+	name = "butterfly knife"
+	desc = "A folding type knife that stores the blade between its two handles when flipped."
+	icon = 'icons/obj/butterfly.dmi'
+	icon_state = "Bflyknife_plain"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/swords_axes.dmi', "right_hand" = 'icons/mob/in-hand/right/swords_axes.dmi')
+	item_state = null
+	hitsound = "sound/weapons/bladeslice.ogg"
+	flags = FPRINT
+	siemens_coefficient = 1
+	sharpness = 1.2
+	sharpness_flags = SHARP_TIP | SHARP_BLADE
+	w_class = W_CLASS_TINY
+	force = 20
+	throwforce = 8
+	throw_speed = 3
+	throw_range = 6
+	w_type = RECYK_METAL
+	melt_temperature = MELTPOINT_STEEL
+	origin_tech = Tc_MATERIALS + "=3"
+	attack_verb = list("slashes", "stabs", "slices", "tears", "rips", "cuts")
+	var/open = FALSE
+	var/bug = null
+	var/knifetype = "plain"
+
+/obj/item/weapon/butterflyknife/attack_self(var/mob/living/L = null)
+	if(!open)
+		unfold()
+		to_chat(L, "You flip \the [src] open.")
+	else
+		fold()
+		to_chat(L, "You flip \the [src] closed.")
+		if(bug)
+			var/turf/T = get_turf(src)
+			var/mob/living/simple_animal/hostile/viscerator/butterfly/X = new bug
+			X.forceMove(T)
+			X.autodie = TRUE
+			if(L && istype(L))
+				handle_faction(X,L)
+			bug = null
+			spawn(250) //The butterfly lives for about 20 seconds and it recharges in 25 seconds.
```

Comment:
I'd say 25 seconds is pretty bad for a spawn. Use process and compare with world time I'd say.